src/core/objective.jl

```diff
@@ -131,31 +163,19 @@ function _objective_min_fuel_and_flow_cost_polynomial_linquad(pm::AbstractPowerM
     for (n, nw_ref) in nws(pm)
         for (i,gen) in nw_ref[:gen]
             pg = sum( var(pm, n, :pg, i)[c] for c in conductor_ids(pm, n) )
-
-            if length(gen["cost"]) == 1
-                gen_cost[(n,i)] = gen["cost"][1]
-            elseif length(gen["cost"]) == 2
-                gen_cost[(n,i)] = gen["cost"][1]*pg + gen["cost"][2]
-            elseif length(gen["cost"]) == 3
-                gen_cost[(n,i)] = gen["cost"][1]*pg^2 + gen["cost"][2]*pg + gen["cost"][3]
-            else
-                gen_cost[(n,i)] = 0.0
-            end
+            gen_cost[(n, i)] = JuMP.@expression(
+                pm.model,
+                sum(v * _pow(pg, d - 1) for (d, v) in enumerate(reverse(gen["cost"]))),
+            )

```

Comment:
So, yes, we can make this a lot simpler. But `^` has weird semantics in JuMP. See the docstring of `_pow` above.

Reply:
re `_pow`, for the case of x^3 or higher, why do you need to cast into a float?  I can imagine that polynomial solvers are going to want integer valued powers.

Reply:
The problem is this method in JuMP:
https://github.com/jump-dev/JuMP.jl/blob/dfc8825811d23c06c076aec3328f971e014f50af/src/operators.jl#L173-L209

It currently casts all integer powers to `QuadExpr`. One fix would be to make this method type unstable. But that could be classified as breaking because it'd return different things.

Reply:
The semantics that I would expect is that it returns the simplest JuMP expression type that can represent the data.  So I am already surprised that it will promote `GenericAffExpr` to `GenericQuadExpr` in the `^0` and `^1` cases.

Can you remind me what the reason is for this? I am guessing that type stability when doing linear algebra operations?

As a non-breaking change I would advocate for expanding the scope of the JuMP definition to return NLExpression for integers great than 2; I would argue this is a non-breaking change because:
1) The code as written does not specify a return type on the function
2) The return type for all existing code, that does not hit the current error block, will have the same return type as before

I might also advocate for a breaking change to make these sematnics not promote to `GenericQuadExpr` in the `^0` and `^1` cases.

Reply:
> Can you remind me what the reason is for this?

It doesn't look like we thought much about it: https://github.com/jump-dev/JuMP.jl/pull/825. Just a legacy weirdness.

Reply:
> As a non-breaking change I would advocate for expanding the scope of the JuMP definition to return NLExpression for integers great than 2

Done

Reply:
This is actually breaking for linear solvers. Re-introduce the `_pow` function.

Reply:
Now updated in JuMP.